packages/wrangler/src/config/config-helpers.ts

```diff
@@ -1,35 +1,127 @@
-	config,
-	script,
-}: {
-	config?: string;
-	script?: string;
-}): string | undefined {
+export function resolveWranglerConfigPath(
+	{
+		config,
+		script,
+	}: {
+		config?: string;
+		script?: string;
+	},
+	options: { useRedirect?: boolean }
+): string | undefined {
 	if (config !== undefined) {
 		return config;
 	}
 
 	const leafPath = script !== undefined ? path.dirname(script) : process.cwd();
-	return findWranglerConfig(leafPath);
+
+	return findWranglerConfig(leafPath, options);
 }
 
 /**
  * Find the wrangler config file by searching up the file-system
  * from the current working directory.
  */
 export function findWranglerConfig(
-	referencePath: string = process.cwd()
+	referencePath: string = process.cwd(),
+	{ useRedirect = false } = {}
 ): string | undefined {
-	return (
+	const userConfigPath =
 		findUpSync(`wrangler.json`, { cwd: referencePath }) ??
 		findUpSync(`wrangler.jsonc`, { cwd: referencePath }) ??
-		findUpSync(`wrangler.toml`, { cwd: referencePath })
-	);
+		findUpSync(`wrangler.toml`, { cwd: referencePath });
+
+	return useRedirect
+		? findRedirectedWranglerConfig(referencePath, userConfigPath)
+		: userConfigPath;
+}
+
+/**
+ * Check whether there is a config file that indicates
+ * @param cwd
+ * @param userConfigPath
+ * @returns
+ */
+function findRedirectedWranglerConfig(
+	cwd: string,
+	userConfigPath: string | undefined
+) {
+	const PATH_TO_DEPLOY_CONFIG = ".wrangler/deploy/config.json";
+	const deployConfigPath = findUpSync(PATH_TO_DEPLOY_CONFIG, { cwd });
+	if (deployConfigPath === undefined) {
+		return userConfigPath;
+	}
+
+	let redirectedConfigPath: string | undefined;
+	const deployConfigFile = readFileSync(deployConfigPath);
+	try {
+		const deployConfig: { configPath?: string } = parseJSONC(
+			deployConfigFile,
+			deployConfigPath
+		);
+		redirectedConfigPath =
+			deployConfig.configPath &&
+			path.resolve(path.dirname(deployConfigPath), deployConfig.configPath);
+	} catch (e) {
+		throw new UserError(
+			dedent`
+				Failed to load the deploy config at ${path.relative(".", deployConfigPath)}
+			`,
+			{ cause: e }
+		);
+	}
+	if (!redirectedConfigPath) {
+		throw new UserError(dedent`
+			A redirect config was found at "${path.relative(".", deployConfigPath)}".
+			But this is not valid - the required "configPath" property was not found.
+			Instead this file contains:
+			\`\`\`
+			${deployConfigFile}
+			\`\`\`
+		`);
+	}
+
+	if (redirectedConfigPath) {
+		if (!fs.existsSync(redirectedConfigPath)) {
+			throw new UserError(dedent`
+				There is a redirect configuration at "${path.relative(".", deployConfigPath)}".
+				But the config path it points to, "${path.relative(".", redirectedConfigPath)}", does not exist.
+			`);
+		}
+		if (userConfigPath) {
+			if (
+				path.join(path.dirname(userConfigPath), PATH_TO_DEPLOY_CONFIG) !==
+				deployConfigPath
+			) {
+				throw new UserError(dedent`
+					Found both a user config file at "${path.relative(".", userConfigPath)}"
```

Comment:
can we please use `configuration` instead of `config` everywhere in our copy, since this is user facing 🙏